plugin/advisor/mysql/mysql_table_naming_convention.go

```diff
@@ -0,0 +1,121 @@
+package mysql
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/plugin/advisor"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/pingcap/tidb/parser/ast"
+)
+
+var (
+	_ advisor.Advisor = (*TableNamingConventionAdvisor)(nil)
+)
+
+func init() {
+	advisor.Register(db.MySQL, advisor.MySQLTableNamingConvention, &TableNamingConventionAdvisor{})
+	advisor.Register(db.TiDB, advisor.MySQLTableNamingConvention, &TableNamingConventionAdvisor{})
+}
+
+// TableNamingConventionAdvisor is the advisor checking for table naming convention.
+type TableNamingConventionAdvisor struct {
+}
+
+// Check checks for table naming convention.
+func (adv *TableNamingConventionAdvisor) Check(ctx advisor.Context, statement string) ([]advisor.Advice, error) {
+	p := newParser()
+
+	root, _, err := p.Parse(statement, ctx.Charset, ctx.Collation)
+	if err != nil {
+		return []advisor.Advice{
+			{
+				Status:  advisor.Error,
+				Title:   "Syntax error",
+				Content: err.Error(),
+			},
+		}, nil
+	}
+	level, err := advisor.NewStatusBySchemaReviewRuleLevel(ctx.Rule.Level)
+	if err != nil {
+		return nil, err
+	}
+	payload, err := api.UnmarshalNamingRulePayload(ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+	format, err := regexp.Compile(payload.Format)
```

Comment:
nit: we can add a payload validation when users set up the policy, but this is also needed for completeness.

Reply:
Maybe we can add the `Valid` function for each rule payload in the `api/schema_system.go`

Reply:
> Maybe we can add the `Valid` function for each rule payload in the `api/schema_system.go`

Yes, doning it now.